clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -71,6 +76,12 @@ public FetchResponseData data() {
         return data;
     }
 
+    /**
+     * From version 3 or later, the authorized and existing entries in `FetchRequest.fetchData` should be in the same order in `responseData`.
+     * Version 13 introduces topic IDs which mean there may be unresolved partitions. If there is any unknown topic ID in the request, the
+     * response will contain a top-level UNKNOWN_TOPIC_ID error and UNKNOWN_TOPIC_ID errors on all the partitions.
+     * We may also return UNKNOWN_TOPIC_ID for a given partition when that partition in the session has a topic ID inconsistent with the broker.
```

Comment:
Should UNKNOWN_TOPIC_ID be FETCH_SESSION_TOPIC_ID_ERROR now?

Reply:
I think I just have the wrong things here completely. There should be INCONSISTENT_TOPIC_ID here as well.

Reply:
I was just thinking about this and realized we may send new error types to clients that may not be able to handle them. I need to review this code again.

Reply:
Ok. Just went through logic for old clients
1. UNKNOWN_TOPIC_ID should not be returned since we won't ever use topic IDs (version 12 requests and below)"
2. FETCH_SESSION_TOPIC_ID_ERROR should not be returned since we won't send version 13+ in a session and will always have zero uuids
3. INCONSISTENT_TOPIC_ID should not be returned, as we won't have topic IDs in the request/session.

The only thing I can think of is downgrading a client while a session is open. I'm not sure if this can happen.